### DISCUSSION TITLE
MariaDB DSN connection from Docker

### DISCUSSION THREAD

**Discussion by masterwishx** [photoprism/photoprism#2197] [Q&A] [answered]
Using Unraid with Photoprism docker also MariaDB docker from linuxserver all work fine.
using method connection : 

 **'DBUSER:DBPASSWORD@tcp(DBIP:DBPORT)/photoprism?parseTime=true'**

should we move to ? : 

      PHOTOPRISM_DATABASE_SERVER: "mariadb:3306"
      PHOTOPRISM_DATABASE_NAME: "photoprism"
      PHOTOPRISM_DATABASE_USER: "photoprism"
      PHOTOPRISM_DATABASE_PASSWORD: "insecure"

Asked becose i cant find DSN Method in last docker-compose.yml only test_DSN

**Comment by lastzero** [comment, marked as answer]
Yes, use the new, individual env variables as the correct DSN for MariaDB requires some additional parameters, so it tends to cause trouble when you make a mistake. Hard to debug.

**Reply by masterwishx** [reply]
Thanks for answer, i think you mean (utf8mb4) like

**PHOTOPRISM_TEST_DSN_MYSQL8: "root:photoprism@tcp(mysql:4001)/photoprism?charset=utf8mb4,utf8&collation=utf8mb4_unicode_ci&parseTime=true"**

i will change to new env vars.

**Reply by lastzero** [reply]
Yes
